sycl/source/detail/graph_impl.cpp

```diff
@@ -912,6 +929,205 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
+         RequiredWGSize[2] != 0);
+    if (EnforcedLocalSize)
+      LocalSize = RequiredWGSize;
+  }
+  // Create update descriptor
+
+  // Storage for individual arg descriptors
+  std::vector<pi_ext_command_buffer_update_memobj_arg_desc_t> MemobjDescs;
+  std::vector<pi_ext_command_buffer_update_pointer_arg_desc_t> PtrDescs;
+  std::vector<pi_ext_command_buffer_update_value_arg_desc_t> ValueDescs;
+  std::vector<pi_ext_command_buffer_update_exec_info_desc_t> ExecInfoDescs;
+  MemobjDescs.reserve(MaskedArgs.size());
+  PtrDescs.reserve(MaskedArgs.size());
+  ValueDescs.reserve(MaskedArgs.size());
+
+  pi_ext_command_buffer_update_kernel_launch_desc UpdateDesc;
+
+  // Collect arg descriptors and fill kernel launch descriptor
+  using sycl::detail::kernel_param_kind_t;
+  for (size_t i = 0; i < MaskedArgs.size(); i++) {
+    auto &NodeArg = MaskedArgs[i];
+    switch (NodeArg.MType) {
+    case kernel_param_kind_t::kind_pointer: {
+      PtrDescs.push_back({static_cast<uint32_t>(NodeArg.MIndex), NodeArg.MPtr});
+    } break;
+    case kernel_param_kind_t::kind_std_layout: {
+      ValueDescs.push_back({static_cast<uint32_t>(NodeArg.MIndex),
+                            static_cast<uint32_t>(NodeArg.MSize),
+                            NodeArg.MPtr});
+    } break;
+    case kernel_param_kind_t::kind_accessor: {
+      sycl::detail::Requirement *Req =
+          static_cast<sycl::detail::Requirement *>(NodeArg.MPtr);
+
+      pi_mem_obj_property MemObjData{};
+
+      switch (Req->MAccessMode) {
+      case access::mode::read: {
+        MemObjData.mem_access = PI_ACCESS_READ_ONLY;
+        break;
+      }
+      case access::mode::write:
+      case access::mode::discard_write: {
+        MemObjData.mem_access = PI_ACCESS_WRITE_ONLY;
+        break;
+      }
+      default: {
+        MemObjData.mem_access = PI_ACCESS_READ_WRITE;
+        break;
+      }
+      }
+      MemObjData.type = PI_KERNEL_ARG_MEM_OBJ_ACCESS;
+      MemobjDescs.push_back(pi_ext_command_buffer_update_memobj_arg_desc_t{
+          static_cast<uint32_t>(NodeArg.MIndex), &MemObjData,
+          static_cast<pi_mem>(Req->MData)});
+
+    } break;
+
+    default:
+      break;
+    }
+  }
+
+  UpdateDesc.num_mem_obj_args = MemobjDescs.size();
+  UpdateDesc.mem_obj_arg_list = MemobjDescs.data();
+  UpdateDesc.num_ptr_args = PtrDescs.size();
+  UpdateDesc.ptr_arg_list = PtrDescs.data();
+  UpdateDesc.num_value_args = ValueDescs.size();
+  UpdateDesc.value_arg_list = ValueDescs.data();
+  UpdateDesc.num_exec_infos = ExecInfoDescs.size();
+  UpdateDesc.exec_info_list = ExecInfoDescs.data();
+
+  UpdateDesc.global_work_offset = &NDRDesc.GlobalOffset[0];
+  UpdateDesc.global_work_size = &NDRDesc.GlobalSize[0];
+  UpdateDesc.local_work_size = LocalSize;
+  UpdateDesc.num_work_dim = NDRDesc.Dims;
+
+  auto Command = MCommandMap[Node];
+  pi_result Res = Plugin->call_nocheck<
+      sycl::detail::PiApiKind::piextCommandBufferUpdateKernelLaunch>(
```

Comment:
General comment that we should put something in the design doc about update, even if it's just adding the UR entry-point to the table of UR entry-points we have, but ideally any other interesting details (probably around buffer accessors since that seemed like the tricky bit)